docs/content/1.guide/7.plugins.md

```diff
@@ -27,6 +27,38 @@ export default defineNitroPlugin((nitroApp) => {
 })
 ```
 
+**Example:** Development Plugin
+
+You can use plugins to run development tasks, such as database migrations.
+
+```ts
+import { migrate } from 'drizzle-orm/better-sqlite3/migrator'
+
+export default defineNitroPlugin(() => {
+  if (process.dev) {
+    migrate(useDb(), { migrationsFolder: './migrations' })
+  }
+})
+```
+
+**Example:** Hooks
+
+You can use plugins to modify the response or run tasks when Nitro is closed.
+
+```ts
+export default defineNitroPlugin((nitro) => {
+  nitro.hooks.hookOnce("close", async () => {
+    // Will run when nitro is closed
+    await new Promise((resolve) => setTimeout(resolve, 500));
+    console.log("Task is done!");
+  });
+  nitro.hooks.hook('render:response', (response) => {
```

Comment:
Hooks are good example. Maybe we can split it into two examples graceful shutdown and modifying response?